command/agent/testagent.go

```diff
@@ -0,0 +1,278 @@
+package agent
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/consul/consul"
+	"github.com/hashicorp/consul/consul/structs"
+	"github.com/hashicorp/consul/logger"
+	"github.com/hashicorp/consul/testutil/retry"
+	"github.com/hashicorp/consul/types"
+	"github.com/hashicorp/consul/version"
+	uuid "github.com/hashicorp/go-uuid"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano()) // seed random number generator
+}
+
+// TempDir defines the base dir for temporary directories.
+var TempDir = os.TempDir()
+
+// TestAgent encapsulates an Agent with a default configuration and
+// startup procedure suitable for testing. It panics if there are errors
+// during creation or startup instead of returning errors. It manages a
+// temporary data directory which is removed after shutdown.
+//
+// todo(fs): do we need the temp data dir if we run in dev mode?
```

Comment:
Yeah, it's probably better coverage-wise to activate non-dev as well. Dev mode bypasses some key places that are worth kicking during our unit tests.